### PR TITLE
Feature/1508 heights map

### PR DESCRIPTION
**Pull request by augusthjerrild**
[IOT-1508]

### CODE REVIEW

File: src/app/gateway/gateway-edit/gateway-edit.component.html [hunk 1]
```diff
@@ -1,153 +1,321 @@
...
 
-    <div class="row mb-5">
-      <div class="form-group mt-3 col-6">
-        <label class="form-label" for="longitude">{{ 'GATEWAY.LONGITUDE' | translate }}</label>
-        <input type="number" class="form-control" id="longitude" name="longitude" [placeholder]="'00'" required
-               [(ngModel)]="gateway.location.longitude" step=".000001" min="-180" max="180" maxlength="12"
-               (keyup)="onCoordinateKey($event)"
-               [ngClass]="{'is-invalid' : formFailedSubmit && errorFields.includes('longitude'), 'is-valid' : formFailedSubmit && !errorFields.includes('longitude')}">
-      </div>
+        <div>
```
Comment (augusthjerrild) [inline src/app/gateway/gateway-edit/gateway-edit.component.html:75]: This is the actual change :-)

File: src/app/shared/components/map/map.component.ts [hunk 2]
```diff
@@ -11,38 +11,62 @@ import {
...
+    private datafordelerName;
+    private heightCurvesName;
+    private ortofotowmts: L.TileLayer;
+    private streetMap: L.TileLayer;
+    private heightsMapWms: L.TileLayer.WMS;
+    private map: L.Map;
     public mapId;
-    private marker;
-    private markers: any;
+    private marker: L.Marker;
```
Comment (fcv-iteratorIt) [inline src/app/shared/components/map/map.component.ts:38]: IMO rename leaflet importet til leaflet

File: src/app/shared/components/map/map.component.ts [hunk 3]
```diff
@@ -11,38 +11,62 @@ import {
...
+import { environment } from "@environments/environment";
 
 @Component({
     selector: "app-map",
     templateUrl: "./map.component.html",
     styleUrls: ["./map.component.scss"],
 })
 export class MapComponent implements OnInit, AfterViewInit, OnChanges, OnDestroy {
-    private map;
+    private satelliteCenterLat = 56.07900207957074;
```
Comment (fcv-iteratorIt) [inline src/app/shared/components/map/map.component.ts:28]: Probably add a commen on what this is, OR move it to a named constant in a constants file

File: src/app/shared/components/map/map.component.ts [hunk 4]
```diff
@@ -86,14 +113,72 @@ export class MapComponent implements OnInit, AfterViewInit, OnChanges, OnDestroy
         }
     }
 
-    changeMarkers() {
+    private changeLayerListener() {
```
Comment (fcv-iteratorIt) [inline src/app/shared/components/map/map.component.ts:116]: rename to: `setupChangeLayerListener`

File: src/app/shared/components/map/map.component.ts [hunk 5]
```diff
@@ -239,18 +324,71 @@ export class MapComponent implements OnInit, AfterViewInit, OnChanges, OnDestroy
...
+                    layers: "dhm_kurve_0_5_m",
+                    version: "1.3.0",
+                    attribution: '&copy; <a href="https://datafordeler.dk/vejledning/brugervilkaar/">Datafordeler</a>',
+                    noWrap: true,
+                    crs: this.datafordelerCrs,
+                } as any //has to be any since transparent has to be "TRUE" but the type expects a boolean
+            );
+
+            // Define layer groups for layer control
+            var baseLayers: L.Control.LayersObject = {
```
Comment (fcv-iteratorIt) [inline src/app/shared/components/map/map.component.ts:364]: Brug const

File: src/app/shared/components/map/map.component.ts [hunk 6]
```diff
@@ -239,18 +324,71 @@ export class MapComponent implements OnInit, AfterViewInit, OnChanges, OnDestroy
...
+                [this.streetViewName]: this.streetMap,
+                [this.datafordelerName ?? "Satellit Danmark"]: this.ortofotowmts,
+            };
+
+            var overlays: L.Control.LayersObject = {
```
Comment (fcv-iteratorIt) [inline src/app/shared/components/map/map.component.ts:369]: brug const

File: src/app/shared/components/map/map.component.ts [hunk 7]
```diff
@@ -239,18 +324,71 @@ export class MapComponent implements OnInit, AfterViewInit, OnChanges, OnDestroy
...
+            const layerControl = L.control.layers(baseLayers, overlays).addTo(this.map);
+
+            //If it's a map with more than 1 marker, then the map change from street view to satelite will fail if zoom is above 16 for some reason.. Therefore, remove the possibility for it to happen.
+            this.map.on("zoomend", () => {
+                if (this.coordinateList) {
+                    const currentZoom = this.map.getZoom();
+
+                    //The layerControl type doesnt have any way to check if map is set. Therefore we have to make it "any".
+                    //If layerControl._map is NOT set, then the picker on the map is not shown.
+                    var tempControl = layerControl as any;
```
Comment (fcv-iteratorIt) [inline src/app/shared/components/map/map.component.ts:382]: brug const

File: src/assets/env.js
Comment (fcv-iteratorIt) [inline src/assets/env.js:1]: Den her bør ikke være i git

**Review by fcv-iteratorIt — CHANGES_REQUESTED** [review]

File: src/assets/i18n/da.json [hunk 8]
```diff
@@ -384,7 +384,7 @@
         "RESPONSIBLE-NAME": "Gateway-ansvarlig",
         "RESPONSIBLE-EMAIL": "Kontaktmail",
         "RESPONSIBLE-PHONE-NUMBER": "Kontakttelefon",
-        "OPERATIONAL-NAME": "Driftansvarlig",
+    "OPERATIONAL-NAME": "Driftsansvarlig",
```
Comment (fcv-iteratorIt) [inline src/assets/i18n/da.json:387]: Der er lige gået et eller andet galt i formatting her

**Review by fcv-iteratorIt — APPROVED** [review]

**Pull request merged**